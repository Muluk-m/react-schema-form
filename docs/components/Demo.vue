<!-- eslint-disable vue/no-v-html -->
<script lang="ts" setup>
import { ref, defineProps } from 'vue';
import { Schema, FormRef } from 'v3-schema-form';
import { Tabs, Tab, Button } from 'vant';
import MarkdownIt from 'markdown-it';
import hljs from 'highlight.js';
import { createDataBySchema } from '../utils';
import 'highlight.js/styles/base16/zenburn.css';

const localMd = MarkdownIt({
  highlight(str, lang) {
    if (lang && hljs.getLanguage(lang)) {
      return hljs.highlight(lang, str).value;
    }

    return '';
  },
});

const props = defineProps<{
  schema: string;
  description: string;
  path: string;
}>();

const decodeSchema = decodeURIComponent(props.schema);
const parseSchema: Schema = JSON.parse(decodeSchema);
const decodeDescription = decodeURIComponent(props.description);
const data = ref(createDataBySchema(parseSchema));
const formRef = ref<FormRef>();
const active = ref('效果');
</script>

<template>
  <div class="v3sf-demo">
    <p v-html="localMd.render(decodeDescription)" />
    <Tabs v-model:active="active">
      <Tab title="效果" class="v3sf-demo__nav">
        <div class="v3sf-demo__view">
          <SchemaForm ref="formRef" v-model="data" :schema="parseSchema" debug />
          <div>
            <Button block round size="small" type="primary" @click="formRef?.validate(false)"
              >测试</Button
            >
          </div>
        </div>
      </Tab>
      <Tab title="Schema">
        <div
          class="v3sf-demo__code"
          v-html="
            localMd.render(`\`\`\`json
${decodeSchema}`)
          "
        />
      </Tab>
      <Tab title="Data">
        <div
          class="v3sf-demo__code"
          v-html="
            localMd.render(`\`\`\`json
${JSON.stringify(data, null, 2)}`)
          "
        />
      </Tab>
    </Tabs>
  </div>
</template>

<style lang="scss" scoped>
.v3sf-demo {
  width: 500px;

  &__view {
    padding: 5px;
    border: 1px solid var(--vp-c-divider);
    border-radius: 5px;
  }

  &__code {
    padding: 5px 15px;
    background-color: var(--vp-code-block-bg);
    border-radius: 5px;
  }
}
</style>
