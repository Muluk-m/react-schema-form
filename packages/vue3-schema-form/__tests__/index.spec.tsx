import { mount } from '@vue/test-utils';
import Demo from './demo';

describe('Vue3SchemaForm', () => {
  const wrapper = mount({
    render() {
      return <Demo />;
    },
  });

  test('should render', async () => {
    expect(wrapper.find('.v3jsf-schema-form').isVisible()).toBe(true);
  });

  test('schema-form/core', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
