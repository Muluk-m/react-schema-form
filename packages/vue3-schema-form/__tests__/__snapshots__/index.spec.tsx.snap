// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue3SchemaForm schema-form/core 1`] = `
"<div class=\\"v3jsf-schema-form\\">
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">字符串1</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">字符串2</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field v3jsf-form-field--column\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">字符串3</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">字符串4</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">数组</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">数值</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">布尔</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text\\">日期</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"van-cell v3jsf-form-field v3jsf-form-field--column\\">
    <!---->
    <div class=\\"van-cell__title\\">
      <div class=\\"v3jsf-field-label\\">
        <div class=\\"v3jsf-field-label__text v3jsf-field-label__text--required\\">文本域</div>
      </div>
      <!---->
    </div>
    <div class=\\"van-cell__value\\">
      <div class=\\"v3jsf-form-field__wrapper\\">
        <div class=\\"v3jsf-form-field__widget\\">
          <div class=\\"v3jsf-form-field__control\\">
            <!---->
          </div>
        </div>
        <div class=\\"v3jsf-form-field__error-message\\"></div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</div>"
`;
